src/elfpy/pricing_models/base.py

```diff
@@ -341,22 +379,28 @@ def check_input_assertions(
     ):
         """Applies a set of assertions to the input of a trading function."""
 
-        assert quantity.amount > 0, (
-            "pricing_models.check_input_assertions: ERROR: " f"expected quantity.amount > 0, not {quantity.amount}!"
+        assert quantity.amount >= 1e-18, (
+            "pricing_models.check_input_assertions: ERROR: "
+            f"expected quantity.amount >= 1e-18, not {quantity.amount}!"
         )
-        assert market_state.share_reserves >= 0, (
+        assert market_state.share_reserves >= 1e-18, (
```

Comment:
I think it makes sense for us to use an enum for this since we repeat it everywhere and it has a meaning ("basically zero")